test/powershell/Modules/Microsoft.PowerShell.Utility/Get-Command.Tests.ps1

```diff
@@ -0,0 +1,68 @@
+
+        BeforeAll {
+            $configFilePath = Join-Path $testdrive "useabbreviationexpansion.json"
+
+            @"
+            {
+                "ExperimentalFeatures": [
+                  "PSUseAbbreviationExpansion"
+                ]
+            }
+"@ > $configFilePath
+
+        }
+
+        It "Valid cmdlets works with name <name> and module <module>" -TestCases @(
+            @{ Name = "i-psdf"; expected = "Import-PowerShellDataFile"; module = $null },
+            @{ Name = "i-psdf"; expected = "Import-PowerShellDataFile"; module = "Microsoft.PowerShell.Utility" },
+            @{ Name = "r-psb" ; expected = "Remove-PSBreakpoint"      ; module = $null },
+            @{ Name = "r-psb" ; expected = "Remove-PSBreakpoint"      ; module = "Microsoft.PowerShell.Utility" }
+        ) {
+            param($name, $expected, $module)
+
+            $command = "Get-Command $name -UseAbbreviationExpansion"
+
+            if ($module) {
+                $command += " -Module $module"
+            }
+
+            $results = pwsh -settingsfile $configFilePath -c "$command | ConvertTo-Json" | ConvertFrom-Json
+            $results | Should -HaveCount 1
+            $results.Name | Should -BeExactly $expected
+        }
+
+        It "Can return multiple results for cmdlets matching abbreviation" {
+            $results = pwsh -settingsfile $configFilePath -c "Get-Command i-C -UseAbbreviationExpansion | ConvertTo-Json" | ConvertFrom-Json
+            $results | Should -HaveCount 3
+            $results[0].Name | Should -BeExactly "Invoke-Command"
+            $results[1].Name | Should -BeExactly "Import-Clixml"
+            $results[2].Name | Should -BeExactly "Import-Csv"
+        }
+
+        It "Will return multiple results for functions matching abbreviation" {
+            $manifestPath = Join-Path $testdrive "test.psd1"
+            $modulePath = Join-Path $testdrive "test.psm1"
+
+            New-ModuleManifest -Path $manifestPath -FunctionsToExport "Get-FooBar","Get-FB" -RootModule test.psm1
+            @"
+            function Get-FooBar { "foobar" }
+            function Get-FB { "fb" }
+"@ > $modulePath
+
+            $results = pwsh -settingsfile $configFilePath -c "Import-Module $manifestPath; Get-Command g-fb -UseAbbreviationExpansion | ConvertTo-Json" | ConvertFrom-Json
+            $results | Should -HaveCount 2
+            $results[0].Name | Should -BeExactly "Get-FB"
+            $results[1].Name | Should -BeExactly "Get-FooBar"
+        }
+
+        It "Non-existing cmdlets returns non-terminating error" {
+            pwsh -settingsfile $configFilePath -c 'try { get-command g-adf -ea stop } catch { $_.fullyqualifiederrorid }' | Should -BeExactly "CommandNotFoundException,Microsoft.PowerShell.Commands.GetCommandCommand"
+        }
+
+        It "No results if wildcard is used" {
+            pwsh -settingsfile $configFilePath -c Get-Command i-psd* -UseAbbreviationExpansion | Should -BeNullOrEmpty
```

Comment:
my assumption is that if there is a command `i-psdf` then this would be found (due to the usual path for get command), is that right? `-UseAbbreviationExpansion` only adds an additional hunt for a command?

Reply:
i suppose that indicates that we should have an additional test to ensure that `-UseAbbreviationExpansion` doesn't negatively effect command lookup

Reply:
The intent of `UseAbbreviationExpansion` is for exact match only (used as an additional hunt with the existing searching logic).  So `i-psdf` will resolve, but `i-psd*` is nonsense.  Can you give me an example of what the additional test would look like?  Not sure I understand.

Reply:
with regard to our current tab completion, a partial match provides all possibilities, so `Get-C<tab>` will match `get-content`, `get-childitem`, etc. It seems to me that we have *increased* the cognitive burden with this change because now the user has to remember how the cmdlet name has been cased which is arbitrary and capricious (IMO). Supporting my example above would allow any cmdlet which case matches `I*-P*S*D*` seems quite reasonable to me since I can't remember how we case things. The more compound the noun the higher the burden.

Reply:
This change doesn't affect the current tab-completion behavior, so `Get-C<tab>` still works as it works today.  The intent of this change isn't to provide another way to search for commands, but to allow tab completion of a shortcut to a well known command.  If there's a desire to support `i-<tab>` as a way to find all commands where the verb starts with `i`, that should be feedback to the associated RFC.